pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -411,15 +411,15 @@ cdef class {{name}}HashTable(HashTable):
         k = kh_get_{{dtype}}(self.table, ckey)
         return k != self.table.n_buckets
 
-    def sizeof(self, deep=False):
+    def sizeof(self, deep: bool = False) -> int:
         """ return the size of my table in bytes """
         overhead = 4 * sizeof(uint32_t) + 3 * sizeof(uint32_t*)
         for_flags = max(1, self.table.n_buckets >> 5) * sizeof(uint32_t)
         for_pairs =  self.table.n_buckets * (sizeof({{dtype}}_t) + # keys
                                              sizeof(Py_ssize_t))   # vals
         return overhead + for_flags + for_pairs
 
-    def get_state(self):
+    def get_state(self) -> dict[str, int]:
```

Comment:
we should be consistent about this pattern, IIRC we use `Dict[str, int]` ? cc @simonjayhawkins ok either way, we should have a rule and convert

Reply:
IIUC there was no objection to using the new syntax, just not ready to convert all existing annotations yet. I recall a discussion about this and I _think_the outcome was that we would wait till closer to 1.3 to use pyupgrade to reduce the potential conflicts with backports but I can't find the relevant discussion. @MarcoGorelli

Reply:
It was #39930 

Removing `--keep-runtime-typing` will upgrade to newer syntax in files with `from __future__ import annotations`

Reply:
ahh yes. it was because mypy in 1.2.x is on an older version. https://github.com/pandas-dev/pandas/pull/39538#pullrequestreview-580875836

nbd. can fixup the backports if necessary to use the older syntax. so imo can do whenever.